flow360/component/flow360_params/solvers.py

```diff
@@ -174,6 +181,22 @@ class NavierStokesSolver(GenericFlowSolverSettings):
 
     model_type: Literal["Compressible"] = pd.Field("Compressible", alias="modelType", const=True)
 
+    low_mach_preconditioner: Optional[bool] = pd.Field(False, alias="lowMachPreconditioner")
+    low_mach_preconditioner_threshold: Optional[NonNegativeFloat] = pd.Field(
+        alias="lowMachPreconditionerThreshold"
+    )
+
+    # pylint: disable=arguments-differ,invalid-name
+    def to_solver(self, params, **kwargs) -> NavierStokesSolver:
+        """
+        Set preconditioner threshold to freestream Mach number
+        """
+
+        if self.low_mach_preconditioner:
+            self.low_mach_preconditioner_threshold = params.freestream.Mach
```

Comment:
What if Mach in freestream is 0? Do we use MachRef

Reply:
No. There's a minimum value specified in the solver. (0.005)